src/linear_algebra/bilinear_form.lean

```diff
@@ -229,29 +280,29 @@ end comp
 
 section lin_mul_lin
 
-variables {R₂ : Type*} [comm_ring R₂] [module R₂ M] {N : Type w} [add_comm_group N] [module R₂ N]
+variables {M₂' : Type*} [add_comm_group M₂'] [module R₂ M₂']
```

Comment:
The stuff in this section can be generalized to a `comm_semiring R₂'` and `semimodule R₂' M₂'`.

Reply:
I made `R₂` a `comm_semiring` and added a new `comm_ring R₃`, which should generalize everything generalizable.